### DISCUSSION TITLE
tutorials/intro-typescript/03-schema-definition-language-sdl

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#140] [course feedback]
# tutorials/intro-typescript/03-schema-definition-language-sdl

Live data, resolvers, and data sources

https://www.apollographql.com/tutorials/intro-typescript/03-schema-definition-language-sdl

**Comment by zerone-jang** [comment]
good!

**Comment by andreashouben** [comment]
The Query type is not really explained in this section.

**Comment by Lsinkololwe** [comment]
Under SDL, I'm unable to complete the code challenge, the text area is disabled and can only see "Bad Request"

**Reply by mabuyo** [reply]
Hey there, thanks for reporting the bug! It should be fixed now if you give it another try.

**Reply by Lsinkololwe** [reply]
Thank you mabuyo, I'm now completing the code challenges

**Comment by HewaliGogoi** [comment]
Is the Query type basically the types Planet and Galaxy? Everything was going good until the first pointer under "Key takeaways" arrived.

**Reply by mabuyo** [reply]
Hi there @HewaliGogoi ! Sorry for the confusion, we actually don't cover the `Query` type until the next lesson so that point shouldn't be there. Hopefully the next lesson clears up your question.
